publication-rest/src/main/java/no/unit/nva/publication/create/CreatePublicationFromImportCandidateHandler.java

```diff
@@ -46,29 +50,30 @@ public CreatePublicationFromImportCandidateHandler(ResourceService importCandida
         this.publicationService = publicationService;
     }
 
+    private static boolean notAuthorizedToProcessImportCandidates(RequestInfo requestInfo) {
+        return !requestInfo.userIsAuthorized(AccessRight.PROCESS_IMPORT_CANDIDATE.name());
+    }
+
     @Override
     protected PublicationResponse processInput(ImportCandidate input, RequestInfo requestInfo, Context context)
-        throws ApiGatewayException {
+            throws ApiGatewayException {
         validateAccessRight(requestInfo);
         validateImportCandidate(input);
         var identifier = input.getIdentifier();
-        return attempt(() -> candidateService.updateImportStatus(identifier, ImportStatus.IMPORTED))
-                   .map(publicationService::autoImportPublication)
-                   .map(PublicationResponse::fromPublication)
-                   .orElseThrow(failure -> rollbackAndThrowException(input));
+        //TODO: refactor this
+        return attempt(() -> candidateService.updateImportStatus(identifier, new Imported(Instant.now(), null, null)))
```

Comment:
What is easier to read and understand  ?
`updateImportStatus(identifier, ImportStatus.IMPORTED)`
or 
`updateImportStatus(identifier, new Imported(Instant.now(), null, null))`

Reply:
The name of the class is stored in the database as a field named `type` with values `Imported`, `NotImported` etc